containers-tests/tests/intset-strictness.hs

```diff
@@ -1,13 +1,31 @@
+{-# LANGUAGE CPP #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
 module Main (main) where
 
 import Prelude hiding (foldl)
 
 import Test.ChasingBottoms.IsBottom
 import Test.Framework (Test, defaultMain, testGroup)
 import Test.Framework.Providers.QuickCheck2 (testProperty)
+#if __GLASGOW_HASKELL__ > 806
+import Test.QuickCheck (Arbitrary (..), Property)
+#endif
 
 import Data.IntSet
 
+#if __GLASGOW_HASKELL__ > 806
+import Utils.NoThunks
+#endif
+
+
+{--------------------------------------------------------------------
+  Arbitrary, reasonably balanced trees
+--------------------------------------------------------------------}
+instance Arbitrary IntSet where
+  arbitrary = do{ xs <- arbitrary
+                ; return (fromList xs)
+                }
```

Comment:
It's a bit unfortunate that this instance is duplicated. We can fix this when we eventually merge the various testsuites into one.

Reply:
I believe this instance needs to be moved behind CPP as well. With GHC < 8.6, CI reports:

```
tests/intset-strictness.hs:24:10: error:
    Not in scope: type constructor or class ‘Arbitrary’
   |
24 | instance Arbitrary IntSet where
   |          ^^^^^^^^^
```